.travis.yml

```diff
@@ -1,47 +1,34 @@
 language: php
-
-addons:
-  apt:
-    packages:
-      - tidy
-
-php:
-  - 5.6
-  - 7.0
-  - 7.1
-
-before_install:
-  - pip install --user codecov
-
-env:
-  global:
-    - DB=MYSQL CORE_RELEASE=4
+dist: precise
 
 matrix:
   include:
-    - php: 7.1
-      env: DB=MYSQL CORE_RELEASE=4 COVERAGE=1
     - php: 5.6
-      env: DB=PGSQL CORE_RELEASE=4
-    - php: 5.6
-      env: DB=SQLITE CORE_RELEASE=4
+      env: DB=MYSQL PHPCS_TEST=1 PHPUNIT_TEST=1
+    - php: 7.0
+      env: DB=PGSQL PHPUNIT_TEST=1
+    - php: 7.1
+      env: DB=MYSQL PHPUNIT_COVERAGE_TEST=1
+    - php: 7.0
+      env: DB=MYSQL
 
 before_script:
   - phpenv rehash
-  - composer self-update || true
-  - git clone git://github.com/silverstripe/silverstripe-travis-support.git ~/travis-support
-  # Install suggested modules in order to maximize test coverage
-  - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss --require "ezyang/htmlpurifier:4.*,silverstripe/cms:^4.0@dev"
-  - cd ~/builds/ss
+  - phpenv config-rm xdebug.ini
+  - echo 'memory_limit = 2G' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - composer validate
+  - composer require silverstripe/recipe-cms 1.0.x-dev --no-update
+  - if [[ $DB == PGSQL ]]; then composer require --no-update silverstripe/postgresql:2.0.x-dev; fi
+  - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+  - echo "SS_BASE_URL=http://localhost:8080/" >> .env
 
-script:
-  # Execute tests with no coverage.  This is the fastest option
-  - "if [ \"$COVERAGE\" != \"1\" ]; then vendor/bin/phpunit comments/tests/; fi"
+  # Bootstrap cms / mysite folder
+  - php ./cms/tests/bootstrap/mysite.php
 
-  # Execute tests with coverage.  Do this for a small
-  - "if [ \"$COVERAGE\" = \"1\" ]; then vendor/bin/phpunit --coverage-clover=coverage.clover comments/tests/; fi"
+script:
+  - if [[ $PHPUNIT_TEST ]]; then vendor/bin/phpunit; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then phpdbg -qrr vendor/bin/phpunit --coverage-clover=coverage.xml; fi
+  - if [[ $PHPCS_TEST ]]; then vendor/bin/phpcs --standard=framework/phpcs.xml.dist src/ tests/ ; fi
 
 after_success:
-    - "if [ \"$COVERAGE\" = \"1\" ]; then mv coverage.clover ~/build/$TRAVIS_REPO_SLUG/; fi"
-    - cd ~/build/$TRAVIS_REPO_SLUG
-    - "if [ \"$COVERAGE\" = \"1\" ]; then travis_retry codecov; fi"
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then bash <(curl -s https://codecov.io/bash) -f coverage.xml; fi
```

Comment:
With codecov there's a codecov.yml file that disables the PR comments which would be good to add

We'll also need to introduce the php-codesniffer composer dev requirement, and add minimum stability dev and prefer stable true to composer.json so this new style of Travis config will run.

I've started work on this here for reference: https://github.com/robbieaverill/silverstripe-comments/commits/pulls/3.0/fix-travis-config